homeassistant/components/fritz/device_tracker.py

```diff
@@ -1,107 +1,196 @@
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for FRITZ!Box component."""
+    _LOGGER.debug("Starting FRITZ!Box device tracker")
+    router = hass.data[DOMAIN][entry.entry_id]
+    tracked = set()
+
+    @callback
+    def update_router():
+        """Update the values of the router."""
+        _async_add_entities(router, async_add_entities, tracked)
+
+    async_dispatcher_connect(hass, router.signal_device_new, update_router)
 
-        if not ip_device:
-            return {}
-        return {"ip": ip_device, "mac": device}
+    update_router()
 
-    def _update_info(self):
-        """Retrieve latest information from the FRITZ!Box."""
-        if not self.success_init:
-            return False
 
-        _LOGGER.debug("Scanning")
-        self.last_results = self.fritz_box.get_hosts_info()
-        return True
+@callback
+def _async_add_entities(router, async_add_entities, tracked):
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    for mac, device in router.devices.items():
+        if mac in tracked:
+            continue
+
+        new_tracked.append(FritzBoxTracker(router, device))
+        tracked.add(mac)
+
+    if new_tracked:
+        async_add_entities(new_tracked)
+
+
+class FritzBoxTracker(ScannerEntity):
+    """This class queries a FRITZ!Box router."""
+
+    def __init__(self, router: FritzBoxTools, device):
+        """Initialize a FRITZ!Box device."""
+        self._router = router
+        self._mac = device.mac_address
+        self._name = device.hostname or DEFAULT_DEVICE_NAME
+        self._active = False
+        self._attrs = {}
+        self._icon = None
+
+    @property
+    def is_connected(self):
+        """Return device status."""
+        return self._active
+
+    @property
+    def name(self):
+        """Return device name."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return device unique id."""
+        return self._mac
+
+    @property
+    def ip_address(self) -> str:
+        """Return the primary ip address of the device."""
+        return self._router.devices[self._mac].ip_address
+
+    @property
+    def mac_address(self) -> str:
+        """Return the mac address of the device."""
+        return self._mac
+
+    @property
+    def hostname(self) -> str:
+        """Return hostname of the device."""
+        return self._router.devices[self._mac].hostname
+
+    @property
+    def source_type(self) -> str:
+        """Return tracker source type."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def device_info(self) -> Dict[str, any]:
+        """Return the device information."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self._mac)},
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "AVM",
+            "model": "FRITZ!Box Tracked device",
```

Comment:
In a future PR if we have the unique id we can register the fritzbox as a device itself and example: https://github.com/home-assistant/core/blob/dev/homeassistant/components/isy994/__init__.py#L228

and add `via_device`